src/charm.py

```diff
@@ -360,6 +372,30 @@ def _resource_reqs_from_config(self) -> ResourceRequirements:
         requests = {"cpu": "100m", "mem": "200Mi"}
         return adjust_resource_requirements(limits, requests, adhere_to_requests=True)
 
+    def _on_get_extra_jwt_issuers(self, event: ActionEvent):
```

Comment:
let's also check if the service is running, otherwise we can't get the envvars
```suggestion
    def _on_get_extra_jwt_issuers(self, event: ActionEvent):
        if not self._oauth2_proxy_service_is_running:
            return event.fail("Service is not ready. Please re-run the action when the charm is active")
```